lib/cisco_node_utils/cmd_ref/service.yaml

```diff
@@ -0,0 +1,30 @@
+# service
+---
+_exclude: [ios_xr]
+
+clear_status:
+  set_value: "clear install status"
+
+delete:
+  set_context: ['terminal dont-ask']
+  set_value: "delete <media><image>"
+
+delete_boot:
+  set_context: ['terminal dont-ask', 'allow delete boot-image']
+  set_value: "delete <media><image>"
+
+image_version:
+  get_command: "show version image <media><image>"
+  get_value: '/NXOS:\s+version\s(\S+)/'
+
+save_config:
+  set_value: "copy running-config startup-config"
+
+upgrade:
+  set_context: ['terminal dont-ask']
+  set_value: "install all nxos <media><image>"
+
+upgraded:
+  kind: boolean
+  get_command: 'show install all status'
```

Comment:
show install all status returns - a detailed log message of all steps done to upgrade/downgrade the device including but not limited to checking image version, md5 and saving config etc. If there are errors, it has logs to the step the image passed. If an image is acceptable and the device can upgrade the last message is ''Finishing the upgrade, switch will reboot". In any case NX-API returns 200. Maybe we can leave this API as is, and add another one that returns the log ? Let me know your thoughts